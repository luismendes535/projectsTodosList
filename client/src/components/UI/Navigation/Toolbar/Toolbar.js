import React from "react";

import classes from "./Toolbar.module.css";
import { NavLink } from "react-router-dom";

const toolbar = props => (
  <header className={classes.Toolbar}>
    <nav className={classes.DesktopOnly}>
      <span>EDirectinsure TODO List</span>
      {props.isAuth ? (
        <ul className={classes.Logout}>
          {props.userName}
          <li className={classes.NavigationItem}>
            <NavLink
              to="/logout"
              activeClassName={classes.active}
              // exact={this.props}
            >
              <i class="large material-icons">exit_to_app</i>
            </NavLink>
          </li>
        </ul>
      ) : null}
    </nav>
  </header>
);

export default toolbar;
