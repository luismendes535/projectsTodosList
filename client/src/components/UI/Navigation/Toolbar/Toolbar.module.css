.Toolbar {
  height: 56px;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  background-color: lightgray;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
  box-sizing: border-box;
  z-index: 90;
  /* Permalink - use to edit and share this gradient: https://colorzilla.com/gradient-editor/#feffff+0,ddf1f9+35,a0d8ef+100;Blue+3D+%2318 */
  background: #feffff; /* Old browsers */
  background: -moz-linear-gradient(
    top,
    #feffff 0%,
    #dce0e2 35%,
    #a0d8ef 100%
  ); /* FF3.6-15 */
  background: -webkit-linear-gradient(
    top,
    #feffff 0%,
    #ddf1f9 35%,
    #9dadb3 100%
  ); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(
    to bottom,
    #feffff 0%,
    #ddf1f9 35%,
    #8e9da1 100%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#feffff', endColorstr='#a0d8ef',GradientType=0 ); /* IE6-9 */
}

.Toolbar nav {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.Toolbar a {
  color: rgb(13, 13, 14);
}

li {
  list-style-type: none;
}
.Logout {
  display: flex;
  align-items: center;
}
