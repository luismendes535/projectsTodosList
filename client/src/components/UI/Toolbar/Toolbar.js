import React from "react";

import classes from "./Toolbar.module.css";
import { NavLink } from "react-router-dom";

const toolbar = ({ isAuth, userName }) => (
  <header className={classes.Toolbar}>
    <nav className={classes.DesktopOnly}>
      <span>EDirectinsure TODO List</span>
      {isAuth ? (
        <ul className={classes.Logout}>
          {userName}
          <li className={classes.NavigationItem}>
            <NavLink to="/logout" activeClassName={classes.active}>
              <i className="large material-icons">exit_to_app</i>
            </NavLink>
          </li>
        </ul>
      ) : null}
    </nav>
  </header>
);

export default toolbar;
