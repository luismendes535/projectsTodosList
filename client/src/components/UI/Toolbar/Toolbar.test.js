import { shallow } from "enzyme";
import React from "react";
import Toolbar from "./Toolbar";

it("expect to render Toolbar component with a NavLink if user is authenticated", () => {
  expect(shallow(<Toolbar isAuth={true} />)).toMatchSnapshot();
});

it("expect to render Toolbar component without any NavLink if user is not authenticated", () => {
  expect(shallow(<Toolbar isAuth={false} />)).toMatchSnapshot();
});
