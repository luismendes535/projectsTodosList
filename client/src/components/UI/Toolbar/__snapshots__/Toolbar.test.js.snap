// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Toolbar component with a NavLink if user is authenticated 1`] = `ShallowWrapper {}`;

exports[`expect to render Toolbar component without any NavLink if user is not authenticated 1`] = `ShallowWrapper {}`;
